BluetoothAdvertisementsKotlin/app/src/main/java/com/example/bluetoothadvertisements/ScannerFragment.kt

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (C) 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bluetoothadvertisements
+
+import android.bluetooth.BluetoothManager
+import android.bluetooth.le.*
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import android.view.*
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.bluetoothadvertisements.adapter.ScannerAdapter
+import com.example.bluetoothadvertisements.databinding.FragmentScannerBinding
+
+private const val TAG = "ScannerFragment"
+private const val SCAN_PERIOD_IN_MILLIS: Long = 90_000
+
+class ScannerFragment : Fragment() {
+
+    private lateinit var binding: FragmentScannerBinding
+    private lateinit var scannerAdapter: ScannerAdapter
+    private var scanCallback: ScanCallback? = null
+    private var bluetoothLeScanner: BluetoothLeScanner? = null
+    private var handler: Handler? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+        retainInstance = true
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentScannerBinding.inflate(inflater, container, false)
+        handler = Handler(Looper.myLooper()!!)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initialize()
+        setupRecyclerViewAdapter()
+
+        /**
+         * Before Android N, Bluetooth didn't need location permission in order to start scanning.
+         * But between Android N and R location permission is required for Bluetooth scanning.
+         * The code below requests location access if it's required and has not yet been granted.
+         */
+        val isLocationPermissionRequired = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
```

Comment:
Could this be the code below? It would be more readable IMO:

        val isLocationPermissionRequired =
           Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && Build.VERSION.SDK_INT <= Build.VERSION_CODES.R

Reply:
yes, except it would become 3 lines. Changing to a range check?